sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -96,6 +95,107 @@ class CollationStringExpressionsSuite
     assert(collationMismatch.getErrorClass === "COLLATION_MISMATCH.EXPLICIT")
   }
 
+  test("INSTR check result on explicitly collated strings") {
+    def testInStr(str: String, substr: String, collationId: Integer, expected: Integer): Unit = {
+      val string = Literal.create(str, StringType(collationId))
+      val substring = Literal.create(substr, StringType(collationId))
+
+      checkEvaluation(StringInstr(string, substring), expected)
+    }
```

Comment:
we shouldn't use unit tests like this, please take a look at the other tests in their appropriate suites:

- CollationSupportSuite.java for unit tests related to newly introduced collation-aware support in CollationSupport.java
- CollationStringExpressionsSuite.scala for e2e SQL tests that should be used in limited quantity

read more in the refactor Jira ticket description notes:
https://issues.apache.org/jira/browse/SPARK-47410